transformer_engine/pytorch/ops/fused/backward_bias_activation.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) 2022-2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+"""Fused backward dbias + dact + quantize."""
+
+from __future__ import annotations
+from typing import Optional
+
+import torch
+
+import transformer_engine_torch as tex
+from transformer_engine.pytorch.fp8 import Recipe
+from transformer_engine.pytorch.ops.basic import Bias
+from transformer_engine.pytorch.ops.basic.activation import (
+    _ActivationOperation,
+    GELU,
+    ReLU,
+)
+from transformer_engine.pytorch.ops.op import (
+    FusedOperation,
+    FusibleOperation,
+    OperationContext,
+)
+from ...utils import clear_tensor_data
+from .._common import maybe_dequantize
+
+_fused_activations = {GELU: tex.dbias_dgelu, ReLU: tex.dbias_drelu}
+_fusible_activations = tuple(_fused_activations.keys())
+
+
+class BackwardBiasActivation(FusedOperation):
+    """Fused backward dbias + dact + quantize
+
+    Uses the next operation's input quantizer.
+
+    """
+
+    def __init__(self, *, bias: Bias, activation: _ActivationOperation):
+        super().__init__((bias, activation))
```

Comment:
`BackwardBiasActivation` is ordering the basic ops in their forward pass order, while `BackwardLinearAdd` is ordering in their backward pass order:
https://github.com/NVIDIA/TransformerEngine/blob/31fc29abe6dedec2477f37a38945f8d9e57b1f14/transformer_engine/pytorch/ops/fused/backward_linear_add.py#L35
I think the forward pass order makes more sense, especially now that we have helper functions to grab input/grad output quantizers, so in the future we should consider modifying `BackwardLinearAdd` to be consistent.

Reply:
Ok, I will work on this in my refactoring PR.